pilot/pkg/model/push_context.go

```diff
@@ -1094,6 +1103,14 @@ func (ps *PushContext) updateContext(
 		ps.AuthzPolicies = oldPushContext.AuthzPolicies
 	}
 
+	if telemetryChanged {
```

Comment:
future: we have things like `skippedCdsConfigs` and equivalent for most types. Since Telemetry only impacts LDS we can probably optimize it a bit to not trigger full pushes